python/pyarrow/tests/test_flight.py

```diff
@@ -412,72 +426,60 @@ def get_token(self):
         return self.token
 
 
-@contextlib.contextmanager
-def flight_server(server_base, *args, **kwargs):
-    """Spawn a Flight server on a free port, shutting it down when done."""
-    auth_handler = kwargs.pop('auth_handler', None)
-    tls_certificates = kwargs.pop('tls_certificates', None)
-    location = kwargs.pop('location', None)
-    try_connect = kwargs.pop('try_connect', True)
-    connect_args = kwargs.pop('connect_args', {})
-
-    if location is None:
-        # Find a free port
-        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        with contextlib.closing(sock) as sock:
-            sock.bind(('', 0))
-            sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-            port = sock.getsockname()[1]
-        ctor = flight.Location.for_grpc_tcp
-        if tls_certificates:
-            ctor = flight.Location.for_grpc_tls
-        location = ctor("localhost", port)
-    else:
-        port = None
-
-    ctor_kwargs = kwargs
-    server_instance = server_base(*args, **ctor_kwargs)
-    # The server instance needs to be initialized before shutdown()
-    # can be called
-    server_instance.init(location,
-                         auth_handler=auth_handler,
-                         tls_certificates=tls_certificates)
-
-    def _server_thread():
-        server_instance.run()
-
-    thread = threading.Thread(target=_server_thread, daemon=True)
-    thread.start()
-
-    # Wait for server to start
-    if try_connect:
-        deadline = time.time() + 5.0
-        client = flight.FlightClient.connect(location, **connect_args)
-        while True:
-            try:
-                list(client.list_flights())
-            except Exception as e:
-                if 'Connect Failed' in str(e):
-                    if time.time() < deadline:
-                        time.sleep(0.025)
-                        continue
-                    else:
-                        raise
-            break
+def test_flight_server_instantiation():
+    locations = [
+        None,
+        'grpc://localhost:0',
+        ('localhost', _find_free_port()),
+    ]
+    for location in locations:
+        server = FlightServer(location)
+        assert isinstance(server.location, flight.Location)
 
-    try:
-        yield location
-    finally:
-        server_instance.shutdown()
-        thread.join(3.0)
+
+@pytest.mark.slow
+def test_server_shutdown_timeout():
+    started = time.time()
+    with SlowStoppingFlightServer():
+        pass
+    elapsed = time.time() - started
+    assert elapsed >= 0.2
+
+    started = time.time()
+    with pytest.raises(TimeoutError):
+        with SlowStoppingFlightServer(shutdown_timeout=0.1):
```

Comment:
Uh, where is `shutdown_timeout` handled? I can't find it anywhere.

Reply:
Outdated, removed this argument.